scheduler/rank.go

```diff
@@ -196,7 +196,7 @@ OUTER:
 		devAllocator.AddAllocs(proposed)
 
 		// Track the affinities of the devices
-		totalDeviceAffinityWeight := 0.0
+		devicesWithAffinities := 0.0
```

Comment:
the final math should be the (sum(matching weight returned by device allocator))/sum(all device weights)